internal/services/dns/dns_tlsa_record_resource.go

```diff
@@ -0,0 +1,322 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package dns
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/dns/2023-07-01-preview/recordsets"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/dns/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+var (
+	_ sdk.Resource           = DnsTLSARecordResource{}
+	_ sdk.ResourceWithUpdate = DnsTLSARecordResource{}
+)
+
+type DnsTLSARecordResource struct{}
+
+func (DnsTLSARecordResource) ModelObject() interface{} {
+	return &DnsTLSARecordResourceModel{}
+}
+
+func (DnsTLSARecordResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return validate.ValidateRecordTypeID(recordsets.RecordTypeTLSA)
+}
+
+func (DnsTLSARecordResource) ResourceType() string {
+	return "azurerm_dns_tlsa_record"
+}
+
+type DnsTLSARecordResourceModel struct {
+	Name              string                        `tfschema:"name"`
+	ResourceGroupName string                        `tfschema:"resource_group_name"`
+	ZoneName          string                        `tfschema:"zone_name"`
+	Ttl               int64                         `tfschema:"ttl"`
+	Record            []DnsTLSARecordResourceRecord `tfschema:"record"`
+	Tags              map[string]string             `tfschema:"tags"`
+	Fqdn              string                        `tfschema:"fqdn"`
+}
+
+type DnsTLSARecordResourceRecord struct {
+	MatchingType        int64  `tfschema:"matching_type"`
+	Selector            int64  `tfschema:"selector"`
+	Usage               int64  `tfschema:"usage"`
+	CertAssociationData string `tfschema:"cert_association_data"`
```

Comment:
Could we spell this out
```suggestion
	CertAssociationData string `tfschema:"certificate_association_data"`
```